### PR TITLE
initial commit of Test-DbaServerName and Repair-DbaServerName

### PR DESCRIPTION
**Pull request by ctrlbold**
I tested replication (and only to the degree that I know how it works) but not mirror breakage.

### CODE REVIEW

**Comment by ctrlbold** [comment]
I don't believe this supports named instances. Test may not either. It needs to. It can't change the \named  name, but can change the server name. This should be done before it's finished (and has to be done before my presentation in early oct. personal goal.)

**Comment by ctrlbold** [comment]
Oh snap, Test-DbaServerName does support named instances! Cool, onto testing Repair-DbaServerName with an instance.

**Comment by ctrlbold** [comment]
Repair support named instances, too. I'm dumm for thinking I was dumm. Testing now.

File: functions/Repair-DbaServerName.ps1 [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+							$options = [System.Management.Automation.Host.ChoiceDescription[]]($yes, $no)
+							$result = $host.ui.PromptForChoice($title, $message, $options, 1)
+							
+							if ($result -eq 1)
+							{
+								Write-Output "Okay, moving on."
+							}
+							else
+							{
+								Write-Output "Removing Mirroring"
```
Comment (dansqldba) [inline functions/Repair-DbaServerName.ps1]: Write-Warning more appropriate? Shouldn't this support WhatIf?

Comment (ctrlbold) [inline functions/Repair-DbaServerName.ps1]: Yes indeed. Fixed.

File: functions/Test-DbaServerName.ps1 [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+				$mirroreddb = $server.Databases | Where-Object { $_.IsMirroringEnabled -eq $true }
+				
+				if ($mirroreddb.count -gt 0)
+				{
+					$dbs = $mirroreddb.name -join ", "
+					$reasons += "Databases are being mirrored: $dbs"
+				}
+				
+				# check for replication
+				$sql = "select name from sys.databases where is_published = 1 or is_subscribed =1 or is_distributor = 1"
```
Comment (dansqldba) [inline functions/Test-DbaServerName.ps1]: This T-SQL fails on SQL 2000, throws an error on screen. Had to add my own Write-Verbose to find out what was causing the error.

Comment (ctrlbold) [inline functions/Test-DbaServerName.ps1]: Good call. Man, I dont even wanna support SQL 2000. Ill make it skip.

File: functions/Test-DbaServerName.ps1 [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+				}
+				catch
+				{
+					if ($NoWarnings -eq $false)
+					{
+						Write-Warning "Can't contact $($server.ComputerNamePhysicalNetBIOS) using Get-Service. This means the script will not be able to automatically restart SQL services."
+					}
+				}
+				
+				if ($rs.count -gt 0)
```
Comment (dansqldba) [inline functions/Test-DbaServerName.ps1]: Something funky is going on at this line which causes bogus warnings about needing to restart SSRS, even on servers where SSRS is not installed.

Comment (dansqldba) [inline functions/Test-DbaServerName.ps1]: Something funky is going on at this line which causes bogus warnings about needing to restart SSRS, even on servers where SSRS is not installed.

**Comment by dansqldba** [comment]
Making $rs null before checking for SSRS services fixed the issues I was having with servers showing they had SSRS when they didn't. Yay!

**Pull request merged**